lib/ash/api/api.ex

```diff
@@ -1958,14 +1958,16 @@ defmodule Ash.Api do
   """
   @callback bulk_update!(
               Enumerable.t(Ash.Resource.record()) | Ash.Query.t(),
-              atom,
+              action :: atom,
               input :: map,
-              Keyword.t()
+              opts :: Keyword.t()
             ) ::
               Ash.BulkResult.t() | no_return
 
   @doc """
-  Updates all items in the provided enumerable or query with the provided input.
+  Updates all items in the provided enumerable or query with the provided input for the action, which
+  is a map, where the keys are the primary keys of the records to update, and the values are maps
```

Comment:
The keys to the input map aren't the primary keys of the records to update. That comes from the enumerable or query in the first argument.

Reply:
Okay, great! What does it mean "that comes"? Like, what could it look like? Will fix when I understand it better.

Reply:
What I mean is that with `bulk_update`, you don't provide primary keys, you provide a query or stream to be updated. For example:

```elixir
Resource
|> Ash.Query.filter(active == true)
|> MyApi.bulk_update(:deactivate)
```

or

```elixir
list_of_records = [%Resource{}, %Resource{}, %Resource{}]

MyApi.bulk_update(list_of_records, :deactivate)
```

Primary keys don't come into the equation in bulk updates or destroys.

Reply:
Okay, so should the action need input, the input map will be the same input for all records in the list/stream?

Reply:
Correct.

Reply:
Gotcha! Fixing :)